ip_country/src/ip_country.rs

```diff
@@ -273,4 +277,34 @@ Line 17: Invalid (AddrParseError(Ip)) IP address in CSV record: 'BOOGA'"#
 .to_string()
         );
     }
+
+    #[test]
+    fn write_error_test() {
+        let mut subject = BitQueue::new();
+        let output = &mut ByteArrayWriter::new();
+
+        subject.add_bits(0b11011, 5);
+        subject.add_bits(0b00111001110011100, 17);
+        subject.add_bits(0b1, 1);
+
+        output.reject_next_write(Error::new(ErrorKind::WriteZero, "Bad file Descriptor"));
+        let result = generate_country_data("ipv4_country_data", subject, output).unwrap_err();
+        assert_eq!(result.kind(), ErrorKind::WriteZero)
+    }
+
+    #[test]
+    fn write_error_from_ip_country() {
+        let stdin = &mut ByteArrayReader::new(PROPER_TEST_DATA.as_bytes());
+        let stdout = &mut ByteArrayWriter::new();
+        let stderr = &mut ByteArrayWriter::new();
+        stdout.reject_next_write(Error::new(ErrorKind::WriteZero, "Bad file Descriptor"));
+
+        let result = ip_country(vec![], stdin, stdout, stderr);
+
+        assert_eq!(result, 1);
+        let stdout_string = String::from_utf8(stdout.get_bytes()).unwrap();
+        let stderr_string = String::from_utf8(stderr.get_bytes()).unwrap();
+        assert_eq!(stderr_string, "Error generating Rust code: Custom { kind: WriteZero, error: \"Bad file Descriptor\" }");
+        assert_eq!(stdout_string, "\n            *** DO NOT USE THIS CODE ***\n            It will produce incorrect results.\n            The process that generated it found these errors:\n\nError generating Rust code: Custom { kind: WriteZero, error: \"Bad file Descriptor\" }\n\n            Fix the errors and regenerate the code.\n            *** DO NOT USE THIS CODE ***\n");
+    }
```

Comment:
Now I'm confused. It seems to me that these two tests with green code test almost the same thing but only at a different level. 

If you agree, please remove one... probably the first placed.